### PR TITLE
Fix packaging

### LINKED ISSUE #2389: mac thinks that napari is named "__main__.py" in bundled app
## 🐛 Bug

My mac seems to think that napari is named "__main__.py" in bundled app

## To Reproduce

1. Launch napari bundled app on mac
2. Click "napari" on menu bar
3. See the menu items "Hide __main__.py" and "Close __main__.py"

image

## Expected behavior

menu items include "Hide napari" and "Close napari", similar to Chrome...

image

## Environment

napari: 0.4.6
Platform: macOS-10.15.7-x86_64-i386-64bit
System: MacOS 10.15.7
Python: 3.8.7 (default, Jan 2 2021, 04:16:43) [Clang 11.0.0 (clang-1100.0.33.17)]
Qt: 5.15.2
PySide2: 5.15.2
NumPy: 1.19.3
SciPy: 1.6.1
Dask: 2021.03.0
VisPy: 0.6.6

OpenGL:
- GL version: 2.1 ATI-3.10.19
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1792x1120, scale 2.0
- screen 2: resolution 2560x1440, scale 1.0

Plugins:
- affinder: 0.1.0
- aicsimageio: 0.2.0
- aicsimageio_delayed: 0.2.0
- brainreg: 0.2.3
- brainreg_standard: 0.2.3
- cellfinder: 0.2.1
- console: 0.0.3
- napari-hdf5-labels-io: 0.2.dev1
- ndtiffs: 0.1.1
- svg: 0.1.4

### PR DESCRIPTION
**Pull request by liu-ziyang**
# Description
* Fixed application named as `__main.py__` instead of napari in menubar listing
* Fixed bundled installer version not set and defaulted to 0.0.1 
* updated documentation in bundling code

## Type of change
- [X] Bug-fix (non-breaking change which fixes an issue)

# References
related discussion also see https://github.com/beeware/briefcase/issues/581
closes #2389 

# How has this been tested?
maunally test in my own pipeline, see https://github.com/ziyangczi/napari/runs/2146011754?check_suite_focus=true

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: bundle.py
Comment (liu-ziyang) [inline bundle.py:118]: confirmed again this cannot be dropped any time soon

File: bundle.py
Comment (liu-ziyang) [inline bundle.py:216]: the briefcase calls need to happen while the pyproject toml is patched, right now it is rolled back too early, resulted in version of the bundle installer set to default 0.0.1

File: .github/workflows/make_bundle.yml
Comment (liu-ziyang) [inline .github/workflows/make_bundle.yml:53]: confirmed latest briefcase fixed the naming issue in menubar, however the latest should ONLY be applied to macos, there is some regression bugs between dmgbuild and briefcase that would cause the import to crash due to https://github.com/al45tair/dmgbuild/blob/master/dmgbuild/core.py#L57 failes on non-mac environments

Comment (liu-ziyang) [reply .github/workflows/make_bundle.yml:53]: to be more specific, on macos the platform.mac_ver()[0].split('.') resulted in empty string therefore int(v) raises value error, and because this is at import level, any package trying to import it would fail. We would have to patch the code at file level (similar to what we are doing now to insert the extra sleep) to skip this value assignment, so that the import would work. I tried this approach and it is working, but too much changes so I didn't bother include in this PR

Comment (tlambert03) [reply .github/workflows/make_bundle.yml:53]: just curious, did you try the updated briefcase on other platforms, and just not include dmgbuild there (since it should only be necessary on macos anyway)

File: bundle.py
Comment (tlambert03) [reply bundle.py:216]: that comment would be useful in the code itself, to prevent regressions

File: .github/workflows/make_bundle.yml
Comment (liu-ziyang) [reply .github/workflows/make_bundle.yml:53]: The briefcase fails on linux for the qt library linking still, so I didn't bother try more things, but theoretically I think this is a great idea

Comment (tlambert03) [reply .github/workflows/make_bundle.yml:53]: ok!  thanks for checking

**Comment by liu-ziyang** [comment]
@tlambert03 if this looks ok to you can you please approve, gonna merge now

File: .github/workflows/make_bundle.yml
Comment (tlambert03) [reply .github/workflows/make_bundle.yml:53]: do you happen to know if we can update windows?  Since linux is the least used bundle... might be nice to just special case linux, rather than mac?

**Review by tlambert03 — APPROVED** [review]
thanks!  I'm approving, but if you have time to check and potentially update briefcase for windows too, that'd be great

**Comment by tlambert03** [comment]
@ziyangczi, I just realized you had clicked merge on this.  can you please not select automerge in the future until the conversation has been completed?  I would have liked to have gotten your thoughts on the above comment.  can you follow up on that in another PR?

**Comment by neuromusic** [comment]
fwiw, I read your approval & comment @tlambert03 as an approval to merge the PR. it's not clear that you were asking for additional work or a response before merging.

**Comment by tlambert03** [comment]
what I'm saying though is he had clicked the automerge button before I even read and approved the PR.  We have a general policy of clicking "approve" even if we have one remaining question, (if the answer to that question isn't critical).  So that the other person can merge at will, when they want to.  usually, that is met with an answer, followed by a merge.   I can, in the future pay closer attention to whether my approval will lead to an instantaneous merge though

**Comment by liu-ziyang** [comment]
> @ziyangczi, I just realized you had clicked merge on this. can you please not select automerge in the future until the conversation has been completed? I would have liked to have gotten your thoughts on the above comment. can you follow up on that in another PR?

I am afraid I don't agree with the proposal here, if you are not comfortable with the code to be added, please do not "approve" the code change. Github allows you to "add comments" or "changes required" to represent your opinion better. If there is work that is out of scope for the PR, like a nice to have feature or followups, they should be discussed separately in a different issue and not blocking the merge of this particular PR. I believe this is standard practice to avoid scope creep on the pull requests.

**Comment by tlambert03** [comment]
> if you are not comfortable with the code to be added, please do not "approve" the code change

yep, as mentioned above, I will pay closer attention to this in the future

File: .github/workflows/make_bundle.yml
Comment (liu-ziyang) [reply .github/workflows/make_bundle.yml:53]: I am afraid I don't have time this week (I am packing up and moving). Feel free to create an issue to track the thought If you would like to.

**Comment by sofroniewn** [comment]
Just catching up on this thread now - for @ziyangczi and anyone coming across this thread I'd like to refer to the [napari core developer guide](https://napari.org/docs/dev/developers/core_dev_guide.html#all-contributors-are-treated-the-same) where we explicitly say:

> It should also be considered best practice to leave a reasonable (24hr) time window after approval before merge to ensure that other core developers have a reasonable chance to weigh in.
>
> We are also an international community, with contributors from many different time zones, some of whom will only contribute during their working hours, others who might only be able to contribute during nights and weekends. It is important to be respectful of other peoples schedules and working habits, even if it slows the project down slightly - we are in this for the long run. In the same vein you also shouldn’t feel pressured to be constantly available or online, and users or contributors who are overly demanding and unreasonable to the point of harassment will be directed to our Code of Conduct. We value sustainable development practices over mad rushes.

This section of the core developer guide was well considered and I'd ask everyone with merge rights to the repository to please respect its spirit and intention.

With that in mind I'd like to propose disabling the new [auto-merge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request) feature from the repo. I know @jni has used it once or twice when trying to fix some tests, but I think the potential harm here outweighs the minor convenience. @jni if you agree with this proposal then I will disable it.

**Comment by jni** [comment]
@sofroniewn very well put. Yes I agree, let's turn it off — it's convenient when waiting on CI but it's not that critical. And anyway I've seen it fire before all the CI is finished (I think some of our tests are not marked "required"), so it was not working properly in that scenario either anyway.

**Comment by sofroniewn** [comment]
I have now disabled auto-merge

**Pull request merged**
